instrumentation/aws-lambda/aws-lambda-events-2.2/library/src/main/java/io/opentelemetry/instrumentation/awslambdaevents/v2_2/internal/ApiGatewayProxyAttributesExtractor.java

```diff
@@ -83,6 +105,37 @@ private static String getHttpUrl(
     return str.length() == 0 ? null : str.toString();
   }
 
+  private static String getV2HttpUrl(APIGatewayV2HTTPEvent request, Map<String, String> headers) {
```

Comment:
this method looks mostly like a copy-paste  from `getHttpUrl` perhaps you could unify these a bit

Reply:
I created an interface to smooth over the differences between the V1 REST and V2 HTTP APIs.  The APIs do not share a common base class so the implementation delegates to the inner field.